### PR TITLE
 New visit after midnight - setting for disabling this characteristic, refs #9156 

### PR DESCRIPTION
**Pull request by peterbo**
Refs #9156

### CODE REVIEW

**Comment by tsteur** [comment]
Hi @peterbo  :) @mattab do you mind having a look? I'm not so much into this. Code wise it looks good to merge :)

File: config/global.ini.php [hunk 1]
```diff
@@ -561,6 +561,10 @@
 ; on a per-request basis;
 pivot_by_filter_default_column_limit = 10
 
+; ONLY CHANGE THIS VALUE WHEN YOU DO NOT USE PIWIK ARCHIVING, SINCE THIS COULD CAUSE PARTIALLY MISSING ARCHIVE DATA
+; Whether to force a new visit at midnight for every visitor. Default 1.
+force_new_visit_midnight = 1
```
Comment (mattab) [inline config/global.ini.php]: maybe we rename to `create_new_visit_after_midnight` and add it after `create_new_visit_when_website_referrer_changes` ?

**Comment by mattab** [comment]
Hi @peterbo FYI the build fails with `Fatal error: Class 'Piwik\Plugins\CoreHome\Tracker\Config' not found in /home/travis/build/piwik/piwik/plugins/CoreHome/Tracker/VisitRequestProcessor.php on line 157`

**Comment by peterbo** [comment]
Sorry for that - I was 100% sure that I did that change, but somehow the patch didn't find its way to GitHub. Now it should be fixed!

**Comment by mattab** [comment]
@peterbo still does not work, please test your patches ;-) PHP Fatal error:  Class 'Piwik\Plugins\CoreHome\Tracker\Config' not found in /home/travis/build/piwik/piwik/plugins/CoreHome/Tracker/VisitRequestProcessor.php on line 157

**Comment by peterbo** [comment]
Ok, now tested in my test-suite. Reminds me of testing even with small patches. Now, all good things come in threes.

**Comment by mattab** [comment]
Cheers @peterbo !

**Comment by MatomoForumNotifications** [comment]
This pull request has been mentioned on **Matomo forums**. There might be relevant details there:

https://forum.matomo.org/t/tracking-of-long-visits/53545/2

**Pull request merged**